libs/vkd3d/resource.c

```diff
@@ -2329,11 +2355,32 @@ HRESULT d3d12_resource_create_committed(struct d3d12_device *device, const D3D12
         else
             allocate_info.heap_flags |= D3D12_HEAP_FLAG_ALLOW_ONLY_NON_RT_DS_TEXTURES;
 
-        allocate_info.vk_image = object->res.vk_image;
+        if (use_dedicated_allocation)
+        {
+            dedicated_info.sType = VK_STRUCTURE_TYPE_MEMORY_DEDICATED_ALLOCATE_INFO;
+            dedicated_info.pNext = NULL;
+            dedicated_info.image = object->res.vk_image;
+            dedicated_info.buffer = VK_NULL_HANDLE;
+            allocate_info.pNext = &dedicated_info;
+        }
+        else
+        {
+            /* We want to allow suballocations and we need the allocation to
+             * be cleared to zero, which only works if we allow buffers */
+            allocate_info.heap_flags &= ~D3D12_HEAP_FLAG_DENY_BUFFERS;
+            allocate_info.flags = VKD3D_ALLOCATION_FLAG_GLOBAL_BUFFER;
```

Comment:
Why FLAG_GLOBAL_BUFFER?

Reply:
`vkd3d_allocate_memory` won't create a buffer otherwise.

Reply:
I guess it's there to request that we want the allocation to have a VkBuffer.